app/pages/lab/visibility.cjsx

```diff
@@ -27,8 +31,22 @@ module.exports = React.createClass
   setterProperty: 'project'
 
   componentDidMount: ->
+    page = if @props.location?.query?.page? then @props.location.query.page else 1
+    @getWorkflowList page
+
+  onPageChange: (page) ->
+    nextQuery = Object.assign {}, @props.location.query, { page }
+    @context.router.push
+      pathname: @props.location.pathname
+      query: nextQuery
+    @getWorkflowList page
+    workflowList = document.getElementById("workflow-list")
```

Comment:
I've been working on trying to use `ref` to target the DOM node that I want for the custom scroll to a specific element, but I'm not getting it to scroll correctly though. 

Should I be using a string, eg`ref="workflow-list"`? or something more like `ref={(node) => {this.workflowList = node; }}`, or `ref={ (node) => @workflowList = ReactDOM.findDOMNode(node) }`?

And in the `.then`, I'm not getting the DOM node to store in a variable. I've been trying things like `node = ReactDOM.findDOMNode(workflowList)` or `ReactDOM.findDOMNode("workflow-list")?.scrollIntoView()`, but this isn't working, and I'm kinda out of ideas for how to get the desired DOM element stored so that I can call `scrollIntoView()` on it. 

If you have a moment @srallen, it'd be great if you could take a look/point me towards how I'm missing the correct syntax.